src/ansible_runner/config/_base.py

```diff
@@ -495,7 +496,7 @@ def wrap_args_for_containerization(self,
         new_args = [self.process_isolation_executable]
         new_args.extend(['run', '--rm'])
 
-        if self.runner_mode == 'pexpect' or getattr(self, 'input_fd', False):
+        if self.runner_mode == 'pexpect' or self.runner_mode == "subprocess" and sys.stdout.isatty():
```

Comment:
Shouldn't you test for `sys.stdin` instead?

Reply:
Because ansible-core use [sys.stdout.isatty](https://github.com/ansible/ansible/blob/7d9889fcb321e458e017342ba7301c06be1a237e/lib/ansible/cli/__init__.py#L503) to check if itself in a tty and use tty mode.